lib/services/local/events.go

```diff
@@ -61,7 +61,17 @@ func (e *EventsService) NewWatcher(ctx context.Context, watch types.Watch) (type
 		case types.KindCertAuthority:
 			parser = newCertAuthorityParser(kind.LoadSecrets)
 		case types.KindToken:
-			parser = newProvisionTokenParser()
+			switch kind.Version {
```

Comment:
This is bad and could lead to duplicate events for ProvisionToken being emitted >:(